JVM crash when running base testsuite test from command line using auth-server-quarkus-embedded

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

When a test is run from command line with profile `auth-server-quarkus-embedded` there is an JVM crash at the end. The test seems to pass correctly. 

### Version

main

### Regression

- [X] The issue is a regression

### Expected behavior

no JVM crash :)

### Actual behavior

```log
[INFO] Tests run: 38, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 78.61 s - in org.keycloak.testsuite.forms.LoginTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 38, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:01 min
[INFO] Finished at: 2024-12-03T10:24:32+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project integration-arquillian-tests-base: 
[ERROR] 
[ERROR] Please refer to /keycloak/testsuite/integration-arquillian/tests/base/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd '/keycloak/testsuite/integration-arquillian/tests/base' && '/usr/lib/jvm/java-17-openjdk-17.0.12.0.7-2.fc39.x86_64/bin/java' '-Djava.awt.headless=true' '-Xms2048m' '-Xmx2048m' '-XX:MetaspaceSize=512m' '-XX:MaxMetaspaceSize=1024m' '--add-opens=java.base/java.lang=ALL-UNNAMED' '--add-opens=java.base/java.util=ALL-UNNAMED' '-Duser.language=en' '-Duser.region=US' '-XX:+ExitOnOutOfMemoryError' '-XX:+HeapDumpOnOutOfMemoryError' '-jar' '/keycloak/testsuite/integration-arquillian/tests/base/target/surefire/surefirebooter-20241203102312423_3.jar' '/keycloak/testsuite/integration-arquillian/tests/base/target/surefire' '2024-12-03T10-23-12_300-jvmRun1' 'surefire-20241203102312423_1tmp' 'surefire_0-20241203102312423_2tmp'
[ERROR] Process Exit Code: 0
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd '/testsuite/integration-arquillian/tests/base' && '/usr/lib/jvm/java-17-openjdk-17.0.12.0.7-2.fc39.x86_64/bin/java' '-Djava.awt.headless=true' '-Xms2048m' '-Xmx2048m' '-XX:MetaspaceSize=512m' '-XX:MaxMetaspaceSize=1024m' '--add-opens=java.base/java.lang=ALL-UNNAMED' '--add-opens=java.base/java.util=ALL-UNNAMED' '-Duser.language=en' '-Duser.region=US' '-XX:+ExitOnOutOfMemoryError' '-XX:+HeapDumpOnOutOfMemoryError' '-jar' '/keycloak/testsuite/integration-arquillian/tests/base/target/surefire/surefirebooter-20241203102312423_3.jar' '/keycloak/testsuite/integration-arquillian/tests/base/target/surefire' '2024-12-03T10-23-12_300-jvmRun1' 'surefire-20241203102312423_1tmp' 'surefire_0-20241203102312423_2tmp'
[ERROR] Process Exit Code: 0
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:714)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:311)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:268)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1334)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1167)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:931)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:342)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:330)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:175)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:76)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:163)
[ERROR] 	at org.apache.maven.buildcache.BuildCacheMojosExecutionStrategy.execute(BuildCacheMojosExecutionStrategy.java:145)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:160)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:827)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:272)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:195)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

### How to Reproduce?

```bash
cd keycloak/testsuite/integration-arquillian/tests/base/
mvn clean install -Pauth-server-quarkus-embedded -Dtest=LoginTest
```

### Anything else?

```
java -version
openjdk version "17.0.12" 2024-07-16
OpenJDK Runtime Environment (Red_Hat-17.0.12.0.7-2) (build 17.0.12+7)
```

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
@@ -194,8 +194,10 @@ public class Picocli {
     }
 
     public void exit(int exitCode) {
-        // hard exit wanted, as build failed and no subsequent command should be executed. no quarkus involved.
-        System.exit(exitCode);
+        if (exitCode != CommandLine.ExitCode.OK && (!Environment.isTestLaunchMode() || isRebuildCheck())) {
+            // hard exit wanted, as build failed and no subsequent command should be executed. no quarkus involved.
+            System.exit(exitCode);
+        }
     }
 
     private int runReAugmentationIfNeeded(List<String> cliArgs, CommandLine cmd, CommandLine currentCommand) {

--- a/services/src/main/java/org/keycloak/services/ServicesLogger.java
+++ b/services/src/main/java/org/keycloak/services/ServicesLogger.java
@@ -30,6 +30,7 @@ import org.keycloak.models.ModelDuplicateException;
 
 import javax.naming.NamingException;
 import java.io.IOException;
+import java.lang.invoke.MethodHandles;
 import java.net.URI;
 
 import static org.jboss.logging.Logger.Level.DEBUG;
@@ -47,7 +48,7 @@ import static org.jboss.logging.Logger.Level.WARN;
 @MessageLogger(projectCode="KC-SERVICES", length=4)
 public interface ServicesLogger extends BasicLogger {
 
-    ServicesLogger LOGGER = Logger.getMessageLogger(ServicesLogger.class, "org.keycloak.services");
+    ServicesLogger LOGGER = Logger.getMessageLogger(MethodHandles.lookup(), ServicesLogger.class, "org.keycloak.services");
 
     @LogMessage(level = INFO)
     @Message(id=1, value="Loading config from %s")

--- a/testsuite/integration-arquillian/tests/base/pom.xml
+++ b/testsuite/integration-arquillian/tests/base/pom.xml
@@ -755,11 +755,6 @@
                     <scope>test</scope>
                 </dependency>
                 <dependency>
-                    <groupId>org.jboss.logmanager</groupId>
-                    <artifactId>log4j-jboss-logmanager</artifactId>
-                    <version>1.3.0.Final</version>
-                </dependency>
-                <dependency>
                     <groupId>net.bytebuddy</groupId>
                     <artifactId>byte-buddy</artifactId>
                     <version>${byte-buddy.version}</version>

